### PR TITLE
Add CI pipeline

### PR DESCRIPTION
**Pull request by tinaselenge**

### CODE REVIEW

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
```
Comment (Frawless) [inline .github/workflows/integration.yaml:21]: I would define matrix strategy where we would define multiple java versions if needed, or it doesn't make sense in that case?

Comment (tinaselenge) [reply .github/workflows/integration.yaml:21]: We currently compile sql-runner targeting java 11 so don't think it matters which java version is installed as long as it's 11 or higher. However, once we support java 17 and compile the sql-runner targeting it, would we want to run it on java 11 to make sure it's backward compatible? That's of course as long as Java 11 is still supported for Flink. In that case, adding multiple java versions to run with would be useful.

**Review by kornys — APPROVED** [review]

File: .github/workflows/integration.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Cache Maven packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
```
Comment (SamBarker) [inline .github/workflows/integration.yaml:30]: We don't need this as a separate step as we have `cache: maven` from `actions/setup-java@v4`

File: .github/workflows/integration.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build with Maven
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: mvn -B clean verify -Pci
```
Comment (SamBarker) [inline .github/workflows/integration.yaml:42]: I don't think we have a `ci` profile in the project poms currently.

File: .github/workflows/integration.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          username: "${{ secrets.IMAGE_REPO_USERNAME }}"
+          password: "${{ secrets.IMAGE_REPO_PASSWORD }}"
+
+      - name: Build Image
+        if: github.ref_name != 'main'
+        uses: docker/build-push-action@v6
+        with:
+          context: flink-sql-runner/
+          platforms: linux/amd64,linux/arm64
+          push: false
```
Comment (SamBarker) [inline .github/workflows/integration.yaml:57]: We specify the `dockerfile` in the build and push action should we be doing that here as well?

**Review by SamBarker — APPROVED** [review]
Thanks @tinaselenge a couple of nits to clean up.

File: .github/workflows/integration.yaml (hunk 1 above)
Comment (tinaselenge) [reply .github/workflows/integration.yaml:21]: Let's keep the base line of java 17 for now.

**Pull request merged**